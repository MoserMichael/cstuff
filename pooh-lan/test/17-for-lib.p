# co-routines / generators as used with for statement

println( ~msg '*** range of numbers ***' )

for i range( ~from 1 ~to 10 )
  println( ~msg 'value ' .. i )
end

println( ~msg '*** for each key and value in table ***' )

days =  {
          'sun'  : 'Sunday' ,
          'mon'  : 'Monday' ,
          'tue'  : 'Tuesday' ,
          'wed'  : 'Wednesday' ,
          'thur' : 'Thursday' ,
          'fri'  : 'Friday' ,
          'sat'  : 'Shabbat'
        }

for [ shortname, fullname ] each( ~table days )
   println( ~msg shortname .. ' => ' .. fullname )
end

println( ~msg '*** for each key and value in table ***' )

for dayval each( ~table days )
    println( ~msg dayval[ 1 ] .. ' => ' .. dayval[ 2 ] )
end

println( ~msg '*** for each key in table ***' )

for val keys( ~table days )
   println( ~msg val )
end

println( ~msg '*** for each value in table ***' )

for val values( ~table days )
   println( ~msg val )
end




