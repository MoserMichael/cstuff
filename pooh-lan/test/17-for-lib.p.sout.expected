*** range of numbers ***
value 1
value 2
value 3
value 4
value 5
value 6
value 7
value 8
value 9
value 10
*** for each key and value in table ***
mon => Monday
fri => Friday
tue => Tuesday
thur => Thursday
sat => Shabbat
sun => Sunday
wed => Wednesday
*** for each key and value in table ***
mon => Monday
fri => Friday
tue => Tuesday
thur => Thursday
sat => Shabbat
sun => Sunday
wed => Wednesday
*** for each key in table ***
mon
fri
tue
thur
sat
sun
wed
*** for each value in table ***
Monday
Friday
Tuesday
Thursday
Shabbat
Sunday
Wednesday
